levant/deploy.go

```diff
@@ -72,6 +81,47 @@ func (c *nomadClient) Deploy(job *nomad.Job, autoPromote int) (success bool) {
 	return
 }
 
+func (c *nomadClient) evaluationInspector(evalID *string) error {
+
+	evalInfo, _, err := c.nomad.Evaluations().Info(*evalID, nil)
+	if err != nil {
+		return err
+	}
+
+	for {
+		switch evalInfo.Status {
+		case nomadStructs.EvalStatusComplete, nomadStructs.EvalStatusFailed, nomadStructs.EvalStatusCancelled:
+			if len(evalInfo.FailedTGAllocs) == 0 {
+				logging.Info("levant/deploy: evaluation %s finished successfully", *evalID)
+				return nil
+			}
+
+			var class, dimension []string
+
+			for group, metrics := range evalInfo.FailedTGAllocs {
+
+				// Iterate the classes and dimensions to generate lists of each failure.
+				for c := range metrics.ClassExhausted {
+					class = append(class, c)
+				}
+				for d := range metrics.DimensionExhausted {
+					dimension = append(dimension, d)
+				}
+
+				logging.Error("levant/deploy: task group %s failed to place %v allocs, failed on %v and exhausted %v",
+					group, metrics.CoalescedFailures+1, class, dimension)
+			}
+
+			return fmt.Errorf("evaluation %v finished with status %s but failed to place allocations",
+				*evalID, evalInfo.Status)
+
+		default:
+			time.Sleep(1 * time.Second)
```

Comment:
With the eval being fetched outside the loop it seems this is the cause for Levant hanging eternally now?

Reply:
Yeah, that is dumb. Will fix.